internal/app/secretless/tcp_proxy_service_test.go

```diff
@@ -0,0 +1,126 @@
+package secretless
+
+import (
+	"fmt"
+	"net"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin"
+)
+
+func TestNewTCPProxyService(t *testing.T) {
+	t.Run("empty constructor arguments result in errors", func(t *testing.T) {
+		_, err := NewTCPProxyService(
+			nil,
+			nil,
+			nil,
+		)
+		assert.Error(t, err)
+	})
+
+	t.Run("non-empty constructor arguments result in no error", func(t *testing.T) {
+		_, err := NewTCPProxyService(
+			new(connectorMock).connect,
+			new(credentialResolverMock).resolveCredentials,
+			new(listenerMock),
+		)
+		assert.NoError(t, err)
+	})
+
+	// TODO: propagates error from resolveCredentials
+	// TODO: resolveCredentials failure propagated
+
+	t.Run("zeroizes credentials from resolveCredentials", func(t *testing.T) {
+		connector := new(connectorMock)
+		credentialResolver := new(credentialResolverMock)
+		listener :=	new(listenerMock)
+
+		ps := tcpProxyService{
+			connector:          connector.connect,
+			resolveCredentials: credentialResolver.resolveCredentials,
+			listener:           listener,
+		}
+		creds := map[string][]byte{
+			"credName": []byte("credValue"),
+		}
+		credentialResolver.On("resolveCredentials").Return(
+			creds,
+			fmt.Errorf("couldn't resolve credentials"))
+
+		err := ps.handleConnection(nil)
+
+		assert.Error(t, err)
+		assert.Equal(
+			t,
+			string(creds["credName"]),
+			strings.Repeat("\x00", len([]byte("credValue"))),
+		)
+	})
+}
+
+func Test_tcpProxyService(t *testing.T) {
+	t.Run("proxy service streams from source to dest", func(t *testing.T) {
+		source, sourceBack := net.Pipe()
+		destination, destinationBack := net.Pipe()
```

Comment:
These names weren't immediately obvious to me, and I would have liked a bit more handholding.  How about:
```go
// This allows us to control and view what happens in the client and backend connections
// that secretless is proxying.  Whatever we write into `clientConnSrc` can be read from
// `clientConn` by the ProxyService, and whatever the ProxyService writes into
// backendConn will be pipe into backendConnDest, so we can verify it.
clientConn, clientConnSrc := net.Pipe()
backendConn, backendConnDest := net.Pipe()
```